### PR TITLE
init

### PR DESCRIPTION
**Pull request by eugtar**

### CODE REVIEW

File: Cli/Program.cs
Comment (saviiskyi) [inline Cli/Program.cs:1]: What is Cli project for? You could make Presentation as a Console App

File: Presentation/Ui/Common/Logger.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿namespace Presentation
+{
+    public class Logger
```
Comment (saviiskyi) [inline Presentation/Ui/Common/Logger.cs:3]: Ambiguous name

File: Presentation/Ui/ArticleUi.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+        public ArticleUi(IArticleService articleService) 
+        {
+            _service = articleService;
+        }
+
+
+        public CreateArticleDto Create()
+        {
+            string title = ReadText("Article title");
```
Comment (saviiskyi) [inline Presentation/Ui/ArticleUi.cs:19]: use var

File: Presentation/Ui/ArticleUi.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            Uri link = new Uri(ReadText("Link to article"));
```
Comment (saviiskyi) [inline Presentation/Ui/ArticleUi.cs:20]: use var

File: Presentation/Ui/Common/Ui.cs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        public Ui() { }
+
+        public string ReadText( string title, bool required = true)
+        {
+            while (true)
+            {
+                Logger.Log("\n(Enter \"exit\" to close the program)");
+                Logger.Log($"{title}: ", false);
+
+                string value = Console.ReadLine() ?? "";
```
Comment (saviiskyi) [inline Presentation/Ui/Common/Ui.cs:14]: use var

File: Presentation/Ui/Common/Ui.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                
+            }
+        }
+
+
+        public int ReadNumber(string title, bool required = true)
+        {
+            while (true)
+            {
+                string number = ReadText(title, required);
```
Comment (saviiskyi) [inline Presentation/Ui/Common/Ui.cs:46]: try to use var in such situations, check the rest of the project for this moment

File: Presentation/Presentation.csproj [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Application\Application.csproj" />
+    <ProjectReference Include="..\Domain\Domain.csproj" />
```
Comment (saviiskyi) [inline Presentation/Presentation.csproj]: Redundant dependency

File: Domain/Abstracts/Repository.cs [hunk 7]
```diff
@@ -0,0 +1,41 @@
+﻿using System.Text.Json;
+
+namespace Domain
+{
+    public abstract class Repository<T> where T : notnull
```
Comment (saviiskyi) [inline Domain/Abstracts/Repository.cs]: Wrong name for this class

File: Domain/Repositories/Dtos/CreateArticleDto.cs [hunk 8]
```diff
@@ -0,0 +1,14 @@
+﻿namespace Domain
+{
+    public class CreateArticleDto
```
Comment (saviiskyi) [inline Domain/Repositories/Dtos/CreateArticleDto.cs]: Are you sure DTOs should be in Domain layer?

File: Domain/Repositories/ArticleRepository.cs [hunk 9]
```diff
@@ -0,0 +1,62 @@
+
+namespace Domain {
+    public class ArticleRepository : Repository<Article>, IArticleRepository
+    {
+        public ArticleRepository() : this("articles") { }
```
Comment (saviiskyi) [inline Domain/Repositories/ArticleRepository.cs]: Are you sure Repositories should be in the Domain layer?

File: Domain/Repositories/ArticleRepository.cs [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+        public ArticleRepository(string name) : base(name) { }
```
Comment (saviiskyi) [inline Domain/Repositories/ArticleRepository.cs]: Can be simplified, without using second constructor

File: Domain/Repositories/ArticleRepository.cs [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+            Article newArticle = new(
+                Guid.NewGuid().ToString(),
+                createArticleDto.Title,
+                createArticleDto.Link,
+                DateTime.Now.TimeOfDay,
+                DateTime.Now.TimeOfDay
+            );
+
+
+            List<Article?> articles = Read();
```
Comment (saviiskyi) [inline Domain/Repositories/ArticleRepository.cs]: What do you wait null here?

File: Domain/Repositories/ArticleRepository.cs [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+            Write([newArticle, .. articles]);
+
+            return newArticle;
+        }
+
+        public void Delete(string id)
+        {
+            List<Article?> articles = Read();
+            if (articles.Count == 0)
```
Comment (saviiskyi) [inline Domain/Repositories/ArticleRepository.cs]: You can use Any() in such situations

File: Domain/Repositories/ArticleRepository.cs [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+            }
+
+            int i = articles.FindIndex(article => article?.Id == id);
+
+            if (i == -1)
+            {
+                new NotImplementedException();
+            }
+
+            articles.RemoveAt(i);
```
Comment (saviiskyi) [inline Domain/Repositories/ArticleRepository.cs]: You can use something like this:
articles.RemoveAll(article => article?.Id == id);

File: Domain/Repositories/ArticleRepository.cs [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+
+        public Article Create(CreateArticleDto createArticleDto)
+        {
+            Article newArticle = new(
+                Guid.NewGuid().ToString(),
+                createArticleDto.Title,
+                createArticleDto.Link,
+                DateTime.Now.TimeOfDay,
+                DateTime.Now.TimeOfDay
+            );
```
Comment (saviiskyi) [inline Domain/Repositories/ArticleRepository.cs]: Are you sure we need to map dto to entity here?

File: Domain/Repositories/ArticleRepository.cs [hunk 15]
```diff
@@ -0,0 +1,62 @@
+
+namespace Domain {
+    public class ArticleRepository : Repository<Article>, IArticleRepository
```
Comment (saviiskyi) [inline Domain/Repositories/ArticleRepository.cs]: Consider using GenericRepository for generic functionality

File: Domain/Enums/EBookFormat.cs [hunk 16]
```diff
@@ -0,0 +1,17 @@
+using System.ComponentModel;
+
+namespace Domain {
+    public enum EBookFormat
+    {
+        [Description(".epub")]
```
Comment (saviiskyi) [inline Domain/Enums/EBookFormat.cs:6]: assign numbers explicitly

File: Domain/Enums/VideoQuality.cs [hunk 17]
```diff
@@ -0,0 +1,15 @@
+using System.ComponentModel;
+
+namespace Domain {
+    public enum VideoQuality
+    {
+        [Description("360p")]
```
Comment (saviiskyi) [inline Domain/Enums/VideoQuality.cs:6]: assign numbers explicitly

**Review by saviiskyi — CHANGES_REQUESTED** [review]

File: Domain/Enums/EBookFormat.cs (hunk 16 above)
Comment (eugtar) [reply Domain/Enums/EBookFormat.cs:6]: Do I need to use description attribute here or it's useless?

Comment (saviiskyi) [reply Domain/Enums/EBookFormat.cs:6]: I have no idea why you added it here, maybe there is some logic behind it

**Pull request closed without merging**